File: frontend/src/types/project.ts
export interface Character {
  id: number;
  name: string;
  role: string;
  description: string;
  appearance: string;
  voice_type: string;
  reference_image_path?: string;
}

export interface Scene {
  id: number;
  scene_number: number;
  chapter_number?: number;
  title: string;
  narration: string;
  emotion: string;
  camera: string;
  image_prompt: string;
  duration_seconds: number;
  character_names: string[];
  image_path?: string;
  audio_path?: string;
}

export interface Chapter {
  id: number;
  chapter_number: number;
  title: string;
  synopsis?: string;
  duration_minutes: number;
  cliffhanger?: string;
  video_path?: string;
  start_scene_number: number;
  end_scene_number: number;
}

export interface Short {
  id: number;
  title: string;
  hook: string;
  start_scene_number: number;
  end_scene_number: number;
  duration_string: string;
  platforms: string[];
  video_path?: string;
}

export interface SeoData {
  youtube?: PlatformSeo;
  tiktok?: PlatformSeo;
  facebook?: PlatformSeo;
  shorts?: PlatformSeo;
  thumbnail_text?: string;
}

export interface PlatformSeo {
  title: string;
  description: string;
  tags: string[];
}

export interface Project {
  id: number;
  title: string;
  synopsis?: string;
  genres: string[];
  style: string;
  duration_minutes: number;
  tone: number;
  mode: string;
  platforms: string[];
  status: 'pendiente' | 'generando' | 'completado' | 'fallido';
  seo_data?: SeoData;
  chapters_mode: boolean;
  chapter_duration_minutes: number;
  total_chapters: number;
  created_at: string;
  scenes: Scene[];
  characters: Character[];
  shorts: Short[];
  chapters: Chapter[];
}

export interface ProjectListItem {
  id: number;
  title: string;
  genres: string[];
  style: string;
  duration_minutes: number;
  total_chapters: number;
  chapters_mode: boolean;
  status: string;
  created_at: string;
  scene_count: number;
}

File: backend/app/schemas/project.py
"""Schemas Pydantic para proyectos."""
from datetime import datetime
from pydantic import BaseModel


class CharacterRead(BaseModel):
    id: int
    name: str
    role: str
    description: str
    appearance: str
    voice_type: str
    reference_image_path: str | None = None

    model_config = {"from_attributes": True}


class SceneRead(BaseModel):
    id: int
    scene_number: int
    chapter_number: int | None = 1
    title: str
    narration: str
    emotion: str
    camera: str
    image_prompt: str
    duration_seconds: int
    character_names: list[str]
    image_path: str | None = None
    audio_path: str | None = None

    model_config = {"from_attributes": True}


class ChapterRead(BaseModel):
    id: int
    chapter_number: int
    title: str
    synopsis: str | None = None
    duration_minutes: int
    cliffhanger: str | None = None
    video_path: str | None = None
    start_scene_number: int
    end_scene_number: int

    model_config = {"from_attributes": True}


class ShortRead(BaseModel):
    id: int
    title: str
    hook: str
    start_scene_number: int
    end_scene_number: int
    duration_string: str
    platforms: list[str]
    video_path: str | None = None

    model_config = {"from_attributes": True}


class ProjectRead(BaseModel):
    id: int
    title: str
    synopsis: str | None
    genres: list[str]
    style: str
    duration_minutes: int
    tone: int
    mode: str
    platforms: list[str]
    status: str
    seo_data: dict | None
    chapters_mode: bool = False
    chapter_duration_minutes: int = 20
    total_chapters: int = 1
    created_at: datetime
    scenes: list[SceneRead] = []
    characters: list[CharacterRead] = []
    shorts: list[ShortRead] = []
    chapters: list[ChapterRead] = []

    model_config = {"from_attributes": True}


class ProjectList(BaseModel):
    id: int
    title: str
    genres: list[str]
    style: str
    duration_minutes: int
    total_chapters: int = 1
    chapters_mode: bool = False
    status: str
    created_at: datetime
    scene_count: int = 0

    model_config = {"from_attributes": True}


class ProjectCreate(BaseModel):
    """No se usa directamente para crear; la generacion crea el proyecto."""
    pass
